wps_jobs/wps_jobs/weather_model_jobs/ecmwf_prediction_processor.py

```diff
@@ -0,0 +1,199 @@
+
+import logging
+from datetime import datetime, timedelta, timezone
+from typing import Dict, List
+from collections import defaultdict
+from wps_shared.utils.time import get_utc_now
+from wps_shared.schemas.stations import WeatherStation
+from wps_shared.db.models.weather_models import ModelRunPrediction, PredictionModelRunTimestamp, WeatherStationModelPrediction
+from wps_shared.db.crud.model_run_repository import ModelRunRepository
+from wps_jobs.weather_model_jobs import ModelEnum, ModelRunProcessResult
+from wps_jobs.weather_model_jobs.machine_learning import StationMachineLearning
+from wps_jobs.weather_model_jobs.utils.interpolate import SCALAR_MODEL_VALUE_KEYS_FOR_INTERPOLATION, construct_interpolated_noon_prediction, interpolate_between_two_points
+
+logger = logging.getLogger(__name__)
+
+
+class ECMWFPredictionProcessor:
+    def __init__(self, stations: List[WeatherStation], model_run_repository: ModelRunRepository, prediction_run: PredictionModelRunTimestamp, machine: StationMachineLearning):        
+        self.prediction_run = prediction_run
+        self.stations = stations
+        self.model_run_repository = model_run_repository
+        self.machine = machine
```

Comment:
Is this necessary? It looks like we instantiate new `StationMachineLearning` instances in `_process_model_run_for_station()`

Reply:
Yeah it's unnecessary, thanks for catching that!